pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -275,13 +275,41 @@ func (r *Reconciler) gcResourceTrackers(logCtx monitorContext.Context, handler *
 			cond.Message = fmt.Sprintf("Waiting for %s to delete. (At least %d resources are deleting.)", waiting[0].DisplayName(), len(waiting))
 		}
 		handler.app.Status.SetConditions(cond)
-		return ctrl.Result{RequeueAfter: baseGCBackoffWaitTime}, r.patchStatus(logCtx, handler.app, phase)
+		return r.result(r.patchStatus(logCtx, handler.app, phase)).requeue(baseGCBackoffWaitTime).ret()
 	}
 	logCtx.Info("GarbageCollected resourcetrackers")
 	if phase == common.ApplicationRendering {
-		return ctrl.Result{}, r.updateStatus(logCtx, handler.app, common.ApplicationRunningWorkflow)
+		return r.result(r.updateStatus(logCtx, handler.app, common.ApplicationRunningWorkflow)).ret()
 	}
-	return ctrl.Result{}, r.patchStatus(logCtx, handler.app, phase)
+	return r.result(r.patchStatus(logCtx, handler.app, phase)).ret()
+}
+
+type reconcileResult struct {
+	time.Duration
+	err error
+}
+
+func (r *reconcileResult) requeue(d time.Duration) *reconcileResult {
+	r.Duration = d
+	return r
+}
+
+func (r *reconcileResult) ret() (ctrl.Result, error) {
+	if r.Duration.Seconds() != 0 {
+		return ctrl.Result{RequeueAfter: r.Duration}, r.err
+	} else if r.err != nil {
+		return ctrl.Result{}, r.err
+	}
+	return ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, nil
```

Comment:
Hello, I wonder why there is no `Result.Requeue` false here. It currently seems that even if the application is completely cleaned up, an Reconcile will still be triggered every 5 minutes.